src/api/fetcher.ts

```diff
@@ -0,0 +1,58 @@
+import CustomError from '@/utils/error';
+
+interface RequestProps {
+  url: string;
+  method: 'GET' | 'POST' | 'DELETE' | 'PATCH' | 'PUT';
+  errorMessage: string;
+  body?: Record<string, string | number>;
+  headers?: Record<string, string>;
+}
+
+type FetchProps = Omit<RequestProps, 'method'>;
+
+const fetcher = {
+  async request({ url, method, errorMessage, body, headers }: RequestProps) {
+    try {
+      const response = await fetch(url, {
+        method,
+        body: body && JSON.stringify(body),
+        headers: headers && headers,
+      });
+      if (response.status === 400)
+        throw new CustomError({ name: 'BAD_REQUEST_ERROR', message: errorMessage });
+      if (response.status === 401)
+        throw new CustomError({ name: 'AUTHORIZED_ERROR', message: errorMessage });
```

Comment:
```suggestion
        throw new CustomError({ name: 'UNAUTHORIZED_ERROR', message: errorMessage });
```

ERROR 앞에 error가 발생한 상황에 대해서 작성하고 있는 것 같아서 ( BAD_REQUEST, NOT_FOUND 등) 이부분도 UNATUTHORIZED 에러라고 작성하는게 통일성있고, 이해가 쉽지 않을까 싶어요~~